dev/archery/archery/integration/datagen.py

```diff
@@ -1401,6 +1437,18 @@ def generate_nested_dictionary_case():
                           dictionaries=[dict0, dict1, dict2])
 
 
+def generate_extension_case():
+    uuid_type = ExtensionType('uuid', 'uuid-serialization',
+                              FixedSizeBinaryField('', 16))
+
+    fields = [
+        ExtensionField('uuids', uuid_type),
```

Comment:
Should we also test dictionary(ext) and ext(storage=dictionary)?

Reply:
dictionary(ext) is not possible as per https://mail-archives.apache.org/mod_mbox/arrow-dev/202004.mbox/%3CCAJPUwMAvxLYxJg_QRgdALSq3XS%2BY0zV_LYwsd6FVNYbA90RAAw%40mail.gmail.com%3E , but I'll try to add ext(dictionary).